src/lib/fetching-util.ts

```diff
@@ -0,0 +1,51 @@
+import { ConstantBackoff, handleAll, retry } from 'cockatiel';
+
+const retryPolicy = retry(handleAll, {
+    maxAttempts: 3, // Try 3 times
+    backoff: new ConstantBackoff(50), // Wait 50ms after each try
+});
+
+async function tryFetching(
+    url: string,
+    method: string,
```

Comment:
Können wir hierfür einen Enum/Type oder so erstellen oder ist das unnötig hier? Oder als type einfach wie in der suggestion
```suggestion
    method: 'GET' | 'POST' | 'PUT' | undefined,
```